src/crc32_test.c

```diff
@@ -0,0 +1,32 @@
+#include "crc32.h"
+#include "test_utils.h"
+#include "utils.h"
+#include "stream.h"
+#include "hex_utils.h"
+
+void test(const char* data, uint32_t expected)
+{
+	stream_t s;
+	stream_initFromHexString(&s, data);
+	const uint8_t* buffer = stream_head(&s);
+
+	uint32_t result = crc32(buffer, stream_availableBytes(&s));
+
+	EXPECT_EQ(result, expected);
+}
+
+void run_crc32_test()
+{
+	test("33", 0x6DD28E9B);
```

Comment:
Ok. Just a note why we usually use TESTCASE macro instead of a function -- if something fails here, you will get "assertion failed on line 15" which isn't exactly helpful ...